test/e2e/parallel/space_test.go

```diff
@@ -480,6 +480,39 @@ func TestSubSpaces(t *testing.T) {
 	})
 }
 
+func TestSubSpaceInheritance(t *testing.T) {
+	// given
+	t.Parallel()
+	// make sure everything is ready before running the actual tests
+	awaitilities := WaitForDeployments(t)
+	hostAwait := awaitilities.Host()
+	memberAwait := awaitilities.Member1()
+
+	t.Run("we create a subSpace, with disable inheritance and do not expect roles and usernames from the parent to be inherited in NSTemplateSet", func(t *testing.T) {
+		// when
+		// we have a parentSpace
+		parentSpace, _, parentSpaceBindings := CreateSpace(t, awaitilities, testspace.WithSpecTargetCluster(memberAwait.ClusterName), testspace.WithTierName("appstudio"))
+		// then
+		// wait until MUR and Space have been provisioned
+		VerifyResourcesProvisionedForSpace(t, awaitilities, parentSpace.Name, UntilSpaceHasStatusTargetCluster(memberAwait.ClusterName))
+		mur, err := hostAwait.WaitForMasterUserRecord(t, parentSpaceBindings.Spec.MasterUserRecord)
+		require.NoError(t, err)
+
+		// when
+		// we also have a subSpace with same tier but with disable inheritance
+		subSpace := CreateSubSpace(t, awaitilities, testspace.WithSpecParentSpace(parentSpace.Name), testspace.WithTierName("appstudio"), testspace.WithSpecTargetCluster(memberAwait.ClusterName), testspace.WithDisableInheritance(true))
+		testsupportsb.CreateSpaceBinding(t, awaitilities.Host(), mur, subSpace, "viewer")
+
+		// wait until subSpace has been provisioned
+		VerifyResourcesProvisionedForSpace(t, awaitilities, subSpace.Name)
+		// check that username and role from parentSpace was not inherited in the subSpace NSTemplateSet
+		spaceBindings, err := hostAwait.ListSpaceBindings(subSpace.Name)
+		require.NoError(t, err)
+		require.Equal(t,len(spaceBindings),1)
+		require.Equal(t,spaceBindings[0].Spec.SpaceRole, "viewer")
```

Comment:
the SpaceBinding inheritance is done "virtually" - we don't replicate the SpaceBindings from the parents. What we do is that Space controller collects all SpaceBindings and updates the NSTemplateSet accordingly. That means that the number of the SpaceBinding CRs will be always the same no matter if the inheritance is disabled or not - what is different is the content of the NSTemplateSet.
```suggestion
		appstudioTier, err := hostAwait.WaitForNSTemplateTier(t, "appstudio")
		require.NoError(t, err)
		memberAwait.WaitForNSTmplSet(t, subSpace.Name, UntilNSTemplateSetHasSpaceRoles(
			SpaceRole(appstudioTier.Spec.SpaceRoles["contributor"].TemplateRef, secondMur.Name)))
```

Reply:
I have modified the test and pushed my changes - please ignore the extra logs, will remove when all working. I seem to get the same roles in the NStemplate regardless of whether disable inheritance is set or not and not sure how to confirm I am using it. this in the NS template:
```
        spec:
          clusterResources:
            templateRef: appstudio-clusterresources-f08f06a-f08f06a
          namespaces:
          - templateRef: appstudio-tenant-76b9d37-76b9d37
          spaceRoles:
          - templateRef: appstudio-admin-85ef82d-85ef82d
            usernames:
            - ee86b9fd-e565-4128-a
          - templateRef: appstudio-contributor-004f85a-004f85a
            usernames:
            - 4a5388d8-3a96-4d7c-8
          tierName: appstudio
```
was thinking with disable inhertance I would just get contributor.  Also test code assumes only one role so if I add a second subspace I will need to allow for 2.

Reply:
the code seems to be OK, no idea why it keeps inheriting the SpaceBinding from the parent space

Reply:
It looks like that the tests are running fine, so it was maybe using some old version of the toolchain-common dependency.